src/session-pool.ts

```diff
@@ -813,9 +817,22 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
       !this.isFull &&
       this._pending + this._pendingPrepare <= this._numWaiters
     ) {
+      let reads = this.options.incStep
+        ? this.options.incStep
+        : DEFAULTS.incStep!;
+      let writes = 0;
+      // Only create 1 write session at a time.
```

Comment:
Is there a specific reason for this? Are we worried that we might overdo the number of write sessions and go over the write fraction?

Reply:
The reasoning behind this is to keep it consistent with the current behavior of the node session pool, which treats the configured fraction for write sessions slightly different from what the Java and Go pools do. The basic working (at the moment) is:
1. Use the write fraction to determine the number of write sessions to create during the initial filling of the session pool. This means that if you configure `min=100` and `write=0.2` you'll end up with an initial session pool with 20 write sessions.
1. The write fraction is __not__ used to determine whether any further write sessions should be created when additional sessions are needed. Instead, write sessions are only created on-demand when the user application requests one, and there is none available in the pool. That is what is happening here.
1. A session that has been marked as a write session, either during the initial filling of the pool, or later because it was created on-demand, will remain a write session during the course of its lifetime. This means that the node session pool will maintain a number of write sessions, even if a write fraction of 0.0 (the default) has been configured, if the user application requires write sessions.

The last point is something we should consider a breaking change if we want to change it, but it is probably better not to change it at all. It is a very nice way of auto-configuring the number of write sessions in the pool.

#1031 further optimizes the way the pool maintains write sessions, but retains the behavior from point 3 above.